ts/WoltLabSuite/Core/Acp/Ui/User/Action/Handler/Dialog/Ban.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Acp/Ui/User/Action/Handler/Dialog
+ * @since       5.5
+ */
+
+import UiDialog from "../../../../../../Ui/Dialog";
+import { DialogCallbackSetup } from "../../../../../../Ui/Dialog/Data";
+import * as Language from "../../../../../../Language";
+import * as Ajax from "../../../../../../Ajax";
+import DatePicker from "../../../../../../Date/Picker";
+
+export class BanDialog {
+  private static instance: BanDialog;
+
+  private banCallback: () => void;
+  private userIDs: number[];
+  private submitElement: HTMLElement;
+  private neverExpiresCheckbox: HTMLInputElement;
+  private reasonInput: HTMLInputElement;
+  private userBanExpiresSettingsElement: HTMLElement;
+  private dialogContent: HTMLElement;
+  public static open(userIDs: number[], callback: () => void): void {
```

Comment:
I suggest to add a newline before this method (unless prettier removes this again?).